The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/10400104580/job/28800076419) [(plain)](https://github.com/rust-lang/rust/commit/2c847698032c2851ecc709ac0631ac1b0b81094b/checks/28800076419/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---
#13 2.654 Building wheels for collected packages: reuse
#13 2.655   Building wheel for reuse (pyproject.toml): started
#13 2.901   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 2.902   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132715 sha256=dfa09868353292d98f811d3efdb0d54d07389e808efc71d68e3b93c514bf8bec
#13 2.902   Stored in directory: /tmp/pip-ephem-wheel-cache-h_gri0kx/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#13 2.905 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#13 3.301 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#13 3.301 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#13 3.818 Collecting virtualenv
#13 3.818 Collecting virtualenv
#13 3.855   Downloading virtualenv-20.26.3-py3-none-any.whl (5.7 MB)
#13 3.929      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 5.7/5.7 MB 79.2 MB/s eta 0:00:00
#13 3.981 Collecting platformdirs<5,>=3.9.1
#13 3.984   Downloading platformdirs-4.2.2-py3-none-any.whl (18 kB)
#13 4.018 Collecting filelock<4,>=3.12.2
#13 4.040 Collecting distlib<1,>=0.3.7
#13 4.043   Downloading distlib-0.3.8-py2.py3-none-any.whl (468 kB)
#13 4.050      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 468.9/468.9 KB 93.6 MB/s eta 0:00:00
#13 4.050      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 468.9/468.9 KB 93.6 MB/s eta 0:00:00
#13 4.132 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 4.319 Successfully installed distlib-0.3.8 filelock-3.15.4 platformdirs-4.2.2 virtualenv-20.26.3
#13 DONE 4.4s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      204736 kB
DirectMap2M:     8183808 kB
DirectMap1G:    10485760 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
    Finished `dev` profile [unoptimized] target(s) in 0.04s
##[endgroup]
downloading https://ci-artifacts.rust-lang.org/rustc-builds-alt/0ba9db87e61adcfd9a978188f61c20d9b423a099/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz
extracting /checkout/obj/build/cache/llvm-0ba9db87e61adcfd9a978188f61c20d9b423a099-true/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz to /checkout/obj/build/x86_64-unknown-linux-gnu/ci-llvm
---
   Compiling tidy v0.1.0 (/checkout/src/tools/tidy)
    Finished `release` profile [optimized] target(s) in 30.76s
##[endgroup]
fmt check
Diff in /checkout/compiler/rustc_middle/src/ty/consts.rs:307:
                 Ok(c) => return Some(c),
                 Err(_) if lit_input.ty.has_aliases() => {
                     // allow the `ty`` to be an alias type, though we cannot handle it here
-                },
+                    return None;
+                }
                 Err(e) => {
                 Err(e) => {
                     tcx.dcx().span_delayed_bug(
                         expr.span,
fmt error: Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt/bin/rustfmt" "--config-path" "/checkout" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/checkout/compiler/rustc_infer/src/infer/mod.rs" "/checkout/compiler/rustc_infer/src/infer/at.rs" "/checkout/compiler/rustc_infer/src/infer/relate/lub.rs" "/checkout/compiler/rustc_infer/src/infer/relate/lattice.rs" "/checkout/compiler/rustc_infer/src/infer/relate/mod.rs" "/checkout/compiler/rustc_infer/src/infer/relate/higher_ranked.rs" "/checkout/compiler/rustc_infer/src/infer/relate/glb.rs" "/checkout/compiler/rustc_infer/src/infer/relate/generalize.rs" "/checkout/compiler/rustc_infer/src/infer/relate/type_relating.rs" "/checkout/compiler/rustc_infer/src/infer/relate/combine.rs" "/checkout/compiler/rustc_infer/src/infer/region_constraints/mod.rs" "/checkout/compiler/rustc_infer/src/infer/region_constraints/leak_check.rs" "/checkout/compiler/rustc_infer/src/infer/free_regions.rs" "/checkout/compiler/rustc_infer/src/infer/lexical_region_resolve/mod.rs" "/checkout/compiler/rustc_infer/src/infer/snapshot/mod.rs" "/checkout/compiler/rustc_infer/src/infer/snapshot/undo_log.rs" "/checkout/compiler/rustc_infer/src/infer/snapshot/fudge.rs" "/checkout/compiler/rustc_infer/src/infer/canonical/canonicalizer.rs" "/checkout/compiler/rustc_infer/src/infer/canonical/mod.rs" "/checkout/compiler/rustc_infer/src/infer/canonical/instantiate.rs" "/checkout/compiler/rustc_infer/src/infer/canonical/query_response.rs" "/checkout/compiler/rustc_infer/src/infer/outlives/mod.rs" "/checkout/compiler/rustc_infer/src/infer/outlives/test_type_match.rs" "/checkout/compiler/rustc_infer/src/infer/outlives/env.rs" "/checkout/compiler/rustc_infer/src/infer/outlives/obligations.rs" "/checkout/compiler/rustc_infer/src/infer/outlives/for_liveness.rs" "/checkout/compiler/rustc_infer/src/infer/outlives/verify.rs" "/checkout/compiler/rustc_infer/src/infer/resolve.rs" "/checkout/compiler/rustc_infer/src/infer/opaque_types/table.rs" "/checkout/compiler/rustc_infer/src/infer/opaque_types/mod.rs" "/checkout/compiler/rustc_infer/src/lib.rs" "/checkout/compiler/rustc_infer/src/traits/engine.rs" "/checkout/compiler/rustc_infer/src/traits/project.rs" "/checkout/compiler/rustc_infer/src/traits/mod.rs" "/checkout/compiler/rustc_infer/src/traits/util.rs" "/checkout/compiler/rustc_infer/src/traits/structural_impls.rs" "/checkout/compiler/rustc_middle/src/lint.rs" "/checkout/compiler/rustc_middle/src/ty/pattern.rs" "/checkout/compiler/rustc_middle/src/ty/typeck_results.rs" "/checkout/compiler/rustc_middle/src/ty/opaque_types.rs" "/checkout/compiler/rustc_middle/src/ty/context/tls.rs" "/checkout/compiler/rustc_middle/src/ty/walk.rs" "/checkout/compiler/rustc_middle/src/ty/context.rs" "/checkout/compiler/rustc_middle/src/ty/sty.rs" "/checkout/compiler/rustc_middle/src/ty/error.rs" "/checkout/compiler/rustc_middle/src/ty/consts.rs" "/checkout/compiler/rustc_middle/src/ty/predicate.rs" "/checkout/compiler/rustc_middle/src/ty/mod.rs" "/checkout/compiler/rustc_middle/src/ty/flags.rs" "/checkout/compiler/rustc_middle/src/ty/fast_reject.rs" "/checkout/compiler/rustc_middle/src/ty/layout.rs" "/checkout/compiler/rustc_middle/src/ty/list.rs" "/checkout/compiler/rustc_middle/src/ty/diagnostics.rs" "/checkout/compiler/rustc_middle/src/ty/generic_args.rs" "/checkout/compiler/rustc_middle/src/ty/adjustment.rs" "/checkout/compiler/rustc_middle/src/ty/closure.rs" "/checkout/compiler/rustc_middle/src/ty/fold.rs" "/checkout/compiler/rustc_middle/src/ty/normalize_erasing_regions.rs" "/checkout/compiler/rustc_middle/src/ty/vtable.rs" "/checkout/compiler/rustc_middle/src/ty/intrinsic.rs" "/checkout/compiler/rustc_middle/src/ty/consts/kind.rs" "/checkout/compiler/rustc_middle/src/ty/consts/valtree.rs" "/checkout/compiler/rustc_middle/src/ty/consts/int.rs" "/checkout/compiler/rustc_infer/src/infer/projection.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Thu Aug 15 07:06:57 UTC 2024
  network time: Thu, 15 Aug 2024 07:06:57 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/compiler/rustc_middle/src/ty/consts.rs
+++ b/compiler/rustc_middle/src/ty/consts.rs
@@ -305,6 +305,10 @@ impl<'tcx> Const<'tcx> {
             // mir.
             match tcx.at(expr.span).lit_to_const(lit_input) {
                 Ok(c) => return Some(c),
+                Err(_) if lit_input.ty.has_aliases() => {
+                    // allow the `ty` to be an alias type, though we cannot handle it here
+                    return None;
+                }
                 Err(e) => {
                     tcx.dcx().span_delayed_bug(
                         expr.span,

--- a/tests/ui/const-generics/adt_const_params/116308.rs
+++ b/tests/ui/const-generics/adt_const_params/116308.rs
@@ -1,5 +1,7 @@
-//@ known-bug: #116308
+//@ check-pass
 #![feature(adt_const_params)]
+
+// Regression test for #116308
 
 pub trait Identity {
     type Identity;

